tt_metal/host_api.hpp

```diff
@@ -127,6 +129,9 @@ KernelHandle CreateKernel(
     const std::variant<CoreCoord, CoreRange, CoreRangeSet> &core_spec,
     const std::variant<DataMovementConfig, ComputeConfig, EthernetConfig> &config);
 
+bool FindPersistentKernel(const std::string &file_name, const std::variant<CoreCoord, CoreRange, CoreRangeSet> &core_spec, const std::variant<DataMovementConfig, ComputeConfig, EthernetConfig> &config);
+KernelHandle AttachPersistentKernel(Program &program, const std::string &file_name, const std::variant<CoreCoord, CoreRange, CoreRangeSet> &core_spec, const std::variant<DataMovementConfig, ComputeConfig, EthernetConfig> &config, Device * device);
```

Comment:
CreateKernel vs AttachPersistentKernel. Is there any semantics reflected in the difference in names?

Reply:
I am a little unclear what determines the "persistent" part.
Do we have a PersistentKernel or a PersistentProgram?
Can Program have some kernels persistent and some not?
Can Persistent Program have non-persistent kernel?